pkg/operator/operator.go

```diff
@@ -26,23 +26,51 @@ type Operator struct {
 	manager manager.Manager
 }
 
-func New(metricsAddr, healthProbeAddr string) (*Operator, error) {
+type OperandConfiguration struct {
+	Image   string
+	Version string
+}
+
+type OperatorConfiguration struct {
+	MetricsAddr     string
+	HealthProbeAddr string
+	Prometheus      stackctrl.PrometheusConfiguration
+	Alertmanager    stackctrl.AlertmanagerConfiguration
+	ThanosSidecar   stackctrl.ThanosConfiguration
+	ThanosQuerier   tqctrl.ThanosConfiguration
+}
+
+func NewOperatorConfiguration(metricsAddr string, healthProbeAddr string, images map[string]string) OperatorConfiguration {
```

Comment:
(suggestion) once a constructor has more than 2/3 arguments, I like to use functional options. It would also avoid passing the map[string]string directly and it would be more explicit IMHO.

```suggestion
func WithPrometheusImage(img string) func(*OperatorConfiguration) {
...
}

func WithThanosImage(img string) func(*OperatorConfiguration) {
...
}

func NewOperatorConfiguration(options ...Option) OperatorConfiguration {
```

Reply:
Nice idea!